x/docs/internal/keeper/querier.go

```diff
@@ -58,3 +59,38 @@ func queryGetSentDocuments(ctx sdk.Context, path []string, keeper Keeper) (res [
 
 	return bz, nil
 }
+
+func queryGetReceivedDocsReceipts(ctx sdk.Context, path []string, keeper Keeper) (res []byte, err sdk.Error) {
+	addr := path[0]
+	address, _ := sdk.AccAddressFromBech32(addr)
+
+	var uuid string
+	if len(path) == 2 {
+		uuid = path[1]
+	}
+
+	var receipts []doctypes.DocumentReceipt
+	var receipt doctypes.DocumentReceipt
+	var bz []byte
+	var err2 error
+
+	//If user wants all his receipts
+	if uuid == "" {
+		receipts = keeper.GetUserReceivedReceipts(ctx, address)
+
+		if receipts == nil {
+			receipts = make([]doctypes.DocumentReceipt, 0)
+		}
+		bz, err2 = codec.MarshalJSONIndent(keeper.cdc, &receipts)
+
+	} else {
+		receipt = keeper.GetReceiptByDocumentUuid(ctx, address, uuid)
+		bz, err2 = codec.MarshalJSONIndent(keeper.cdc, &receipt)
+	}
+
```

Comment:
I think the above `if - else` should in both case return a `types.DocumentReceipts` instance. Either if we are retrieving the whole receipts list, or the receipts for a single document, we should be able to get a list and not a single receipt. 

This is better explained into the previous comment.